src/main/java/com/algorand/algosdk/transaction/Transaction.java

```diff
@@ -950,6 +961,7 @@ public AssetParams(
 			   Address freeze,
 			   Address clawback) {
             if(assetTotal != null) this.assetTotal = assetTotal;
+            if(assetDecimals != null) this.assetDecimals = assetDecimals;
```

Comment:
null check won't be needed here once assetDecimals is converted to a primitive data type.

Reply:
Please check that the number is indeed between 0 and 19

Reply:
what about this check?